### DISCUSSION TITLE
Transition effect on page navigation

### DISCUSSION THREAD

**Discussion by shabshabar** [wojtekmaj/react-pdf#919] [Q&A]
Does this library support  transition effect on page navigation, what I mean to say is from going page1 to page 2 can we add a transition effect.

**Comment by wojtekmaj** [comment]
Assuming by "Navigation" you mean rendering page 2 where page 1 was previously rendered:

Yes. The only tricky thing here is to keep the previous page rendered until the next one is fully rendered. The latter problem has been discussed in https://github.com/wojtekmaj/react-pdf/issues/875.

Assuming by "Navigation" you mean rendering page 2 below page 1:

Yes. If you have some external navigation between pages, you can use `window.scrollTo({ (…), behavior: 'smooth' });` to smoothly transition to the page you selected.

Assuming by "Navigation" you mean something like page turning animation: see https://github.com/wojtekmaj/react-pdf/issues/741.
